release/k3s/k3s.go

```diff
@@ -0,0 +1,568 @@
+	"github.com/google/go-github/v39/github"
+	"github.com/rancher/ecm-distro-tools/repository"
+	"github.com/sirupsen/logrus"
+	"github.com/urfave/cli"
+	ssh2 "golang.org/x/crypto/ssh"
+	"gopkg.in/yaml.v2"
+)
+
+const (
+	k8sUpstreamURL     = "https://github.com/kubernetes/kubernetes"
+	rancherRemote      = "k3s-io"
+	k8sRancherURL      = "git@github.com:k3s-io/kubernetes.git"
+	k8sUserURL         = "git@github.com:user/kubernetes.git"
+	k3sUpstreamRepoURL = "https://github.com/k3s-io/k3s"
+	gitconfig          = `
+	[safe]
+	directory = /home/go/src/kubernetes
+	[user]
+	email = %email%
+	name = %user%
+	`
+	dockerDevImage = `
+	FROM %goimage%
+	RUN apk add --no-cache bash git make tar gzip curl git coreutils rsync alpine-sdk
+	`
+)
+
+type Release struct {
+	OldK8SVersion string `json:"old_k8s_version"`
+	NewK8SVersion string `json:"new_k8s_version"`
+	OldK8SClient  string `json:"old_k8s_client"`
+	NewK8SClient  string `json:"new_k8s_client"`
+	OldK3SVersion string `json:"old_k3s_version"`
+	NewK3SVersion string `json:"new_k3s_version"`
+	ReleaseBranch string `json:"release_branch"`
+	Workspace     string `json:"workspace"`
+	Handler       string `json:"handler"`
+	Email         string `json:"email"`
+	Token         string `json:"token"`
+}
+
+func NewReleaseFromConfig(c *cli.Context) (*Release, error) {
+	configPath := c.String("config")
+	var release *Release
+	if configPath == "" {
+		return nil, errors.New("error: config file required")
+	}
+
+	configData, err := os.ReadFile(configPath)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := json.Unmarshal(configData, &release); err != nil {
+		return nil, err
+	}
+	return release, nil
+}
+
+// SetupK8sRemotes will clone the kubernetes upstream repo and proceed with setting up remotes
+// for rancher and user's forks, then it will fetch branches and tags for all remotes
+func (r *Release) SetupK8sRemotes(ctx context.Context, ghClient *github.Client) error {
+	k3sDir := filepath.Join(r.Workspace, "kubernetes")
+	_, err := os.Stat(r.Workspace)
+	if err != nil {
+		if os.IsNotExist(err) {
+			if err := os.MkdirAll(r.Workspace, 0755); err != nil {
+				return err
+			}
+		} else {
+			return err
+		}
+	}
+	// clone the repo
+	repo, err := git.PlainClone(k3sDir, false, &git.CloneOptions{
+		URL:             k8sUpstreamURL,
+		Progress:        os.Stdout,
+		InsecureSkipTLS: true,
+	})
+
+	if err != nil {
+		if err == git.ErrRepositoryAlreadyExists {
+			logrus.Warnf("Repository already exists")
+			repo, err = git.PlainOpen(k3sDir)
+			if err != nil {
+				return err
+			}
+		} else {
+			return err
+		}
+	}
+	gitAuth, err := getAuth("")
+	if err != nil {
+		return err
+	}
+	if err := repo.Fetch(&git.FetchOptions{
+		RemoteName:      "origin",
+		Progress:        os.Stdout,
+		Tags:            git.AllTags,
+		InsecureSkipTLS: true,
+	}); err != nil {
+		if err != git.NoErrAlreadyUpToDate {
+			return err
+		}
+	}
+
+	_, err = repo.CreateRemote(&config.RemoteConfig{
```

Comment:
Can you scope lock the calls where err is the only value we're care about? ie if _, err :=